FirebaseAdmin/FirebaseAdmin/Auth/ImportUserRecordArgs.cs

```diff
@@ -0,0 +1,210 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+
+namespace FirebaseAdmin.Auth
+{
+    /// <summary>
+    /// Represents a user account to be imported to Firebase Auth via the
+    /// <a cref="o:FirebaseAuth.ImportUsersAsync">FirebaseAuth.ImportUsersAsync</a> API. Must contain at least a
+    /// uid string.
+    /// </summary>
+    public sealed class ImportUserRecordArgs
+    {
+        /// <summary>
+        /// Key name for custom attributes.
+        /// </summary>
+        private const string CustomAttributes = "customAttributes";
+
+        /// <summary>
+        /// Gets or sets the uid of the user.
+        /// </summary>
+        public string Uid { get; set; }
+
+        /// <summary>
+        /// Gets or sets the email address of the user.
+        /// </summary>
+        public string Email { get; set; }
+
+        /// <summary>
+        /// Gets or sets if the email was verified, null signifies that it was not specified.
+        /// </summary>
+        public bool? EmailVerified { get; set; }
+
+        /// <summary>
+        /// Gets or sets the display name of the user.
+        /// </summary>
+        public string DisplayName { get; set; }
+
+        /// <summary>
+        /// Gets or sets phone number of the user.
+        /// </summary>
+        public string PhoneNumber { get; set; }
+
+        /// <summary>
+        /// Gets or sets the photo url.
+        /// </summary>
+        public string PhotoUrl { get; set; }
+
+        /// <summary>
+        /// Gets or sets the disabled value, null signifies that it was not specified.
+        /// </summary>
+        public bool? Disabled { get; set; }
+
+        /// <summary>
+        /// Gets or sets the UserMetadata.
+        /// </summary>
+        public UserMetadata UserMetadata { get; set; }
+
+        /// <summary>
+        /// Gets or sets the password hash.
+        /// </summary>
+        public byte[] PasswordHash { get; set; }
+
+        /// <summary>
+        /// Gets or sets the password salt.
+        /// </summary>
+        public byte[] PasswordSalt { get; set; }
+
+        /// <summary>
+        /// Gets or sets the user providers.
+        /// </summary>
+        public IEnumerable<UserProvider> UserProviders { get; set; }
+
+        /// <summary>
+        /// Gets or sets the custom claims.
+        /// </summary>
+        public IReadOnlyDictionary<string, object> CustomClaims { get; set; }
+
+        internal bool HasPassword()
+        {
+            return this.PasswordHash != null;
+        }
+
+        internal ImportUserRequest ToImportUserRequest()
+        {
+            return new ImportUserRequest(this);
+        }
+
+        private static string UrlSafeBase64Encode(byte[] bytes)
+        {
+            var base64Value = Convert.ToBase64String(bytes);
+            return base64Value.TrimEnd('=').Replace('+', '-').Replace('/', '_');
+        }
+
+        internal sealed class ImportUserRequest
```

Comment:
Let's find a different name. You also have an `UserImportRequest` class.

Reply:
Would `ImportUserRecordArgsRequest` be too long?